### PR TITLE
Documentation Parser: Extended HTML Files Generator

### PR DESCRIPTION
**Pull request by BinarySoftware**
### Pull Request Description
This PR adds an important element to Doc Parser. As it is now fully working and fully integrated with Enso Parser, it can do it's job by finally generating HTML file to be displayed in Enso Studio.

Here you can see an example file generated with Doc Parser, thanks to extended integration of it with Enso Parser. It contains many tokens that you have possibility to create thanks to Doc Parser, so it may not look great, but it shows you how many things you can really create, and with proper data it'll look brilliant
Zrzut ekranu 2019-10-17 o 14 35 51

I think that I can safely say that Doc Parser is a better version of Markdown, isn't it?

### Important Notes
This PR is WIP, so I'm open to any suggestions

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/luna/enso/blob/master/doc/scala-style-guide.md), [Java](https://github.com/luna/enso/blob/master/doc/java-style-guide.md) or [Haskell](https://github.com/luna/enso/blob/master/doc/haskell-style-guide.md) style guides as appropriate.
- [x] All code has been tested where possible.

### CODE REVIEW

File: Syntax/specialization/src/main/scala/org/enso/syntax/text/Parser.scala [hunk 1]
```diff
@@ -307,26 +307,40 @@ object Main extends App {
...
-              |sub x y = x - y
-              |
-              |## Foo bar baz
-              |   bax
-              |def Maybe a
-              |    ## test attached to Just
-              |    def Just val:a
-              |    def Nothing
-              |""".stripMargin
+  val inp =
```
Comment (iamrecursion) [inline Syntax/specialization/src/main/scala/org/enso/syntax/text/Parser.scala]: There may be some issues with committing this from Apple's documentation. I don't know what the licensing situation is. This should be converted to our own docs.

Comment (BinarySoftware) [reply Syntax/specialization/src/main/scala/org/enso/syntax/text/Parser.scala]: Okay, written my own, used apple's while testing with @wdanilo

File: Syntax/specialization/src/main/scala/org/enso/syntax/text/DocParser.scala [hunk 2]
```diff
@@ -332,24 +262,261 @@ object DocParserRunner {
...
-  //////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+//// Doc Parser HTML Generator /////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+
+/**
+  * This is Doc Parser HTML Generator.
+  *
+  * Essentially it enables Doc Parser to create pretty HTML files from documented
+  * code.
```
[the remark this answers is not in the document]
Comment (BinarySoftware) [reply Syntax/specialization/src/main/scala/org/enso/syntax/text/DocParser.scala]: Fixed?

[the remark this answers is not in the document]
Comment (iamrecursion) [reply Syntax/specialization/src/main/scala/org/enso/syntax/text/DocParser.scala]: Just set up a column ruler in your IDE/Editor!

**Comment by iamrecursion** [comment]
@wdanilo you need to review this too.

**Comment by BinarySoftware** [comment]
As discussed with @wdanilo , CSS -> Sass

**Pull request merged**